buildenv/jenkins/jobs/infrastructure/signedOffByCheck

```diff
@@ -0,0 +1,100 @@
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+def FAIL = false
+def WRONG_COMMIT = false
+def SRC_REPO = 'https://github.com/${ghprbGhRepository}.git'
+def HASHES = '###################################'
+def COMMITS = []
+def BAD_COMMITS = []
+
+timeout(time: 6, unit: 'HOURS') {
+    stage('SignedOffBy Check') {
+        node ('worker') {
+            timestamps {
+                try {
+                    checkout changelog: false, poll: false,
+                        scm: [$class: 'GitSCM',
+                            branches: [[name: ghprbActualCommit]],
+                            doGenerateSubmoduleConfigurations: false,
+                            extensions: [[$class: 'CloneOption',
+                                            depth: 0,
+                                            noTags: false,
+                                            reference: '/home/jenkins/openjdk_cache',
+                                            shallow: false]],
+                            submoduleCfg: [],
+                            userRemoteConfigs: [[refspec: "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*",
+                                                    url: SRC_REPO]]]
+
+                    COMMITS = sh(
+                         script : "git log origin/${ghprbTargetBranch}..HEAD",
+                         returnStdout: true
+                         ).trim()
+                    if (COMMITS == "") {
+                        echo "There are no commits to check for signoffby message"
+                    } else {
+                        def COMMIT_RAW = COMMITS.split("(?=commit [a-f0-9]{40}+\n(Merge: [a-f0-9 ]+)*\n*Author: .*\nDate:   .*+\n)");
+                        COMMIT_RAW.each { COMMIT ->
+                            if (COMMIT != "") {
+                                def splitCommit = COMMIT.split("\\r?\\n")
+                                def lastBlankLine = 0
+                                for (int i = splitCommit.size() - 1; i >= 0; i--) {
+                                    if ( splitCommit[i] ==~ /^\s*$/ ) {
+                                        lastBlankLine = i
+                                        break  
+                                    }                                                               
+                                }
+                                for (int j = splitCommit.size() - 1; j >= lastBlankLine; j-- ) {
```

Comment:
I think `>= lastBlankLine` is 1 more than needed. I think we will end up checking the blank line on the last iteration.

Reply:
fixed